test/JuMP_sparse_linear_mpc.jl

```diff
@@ -0,0 +1,76 @@
+using Ipopt, JuMP, Random, LinearAlgebra
+
+
+
+"Build a JuMP model to minimize 1/2 sum(x^T Q x for i in 1:nt) + 1/2 sum(u^T R u for i in 1:nt-1)
+subject to x(t+1) = Ax(t) + Bu(t) and lvar <= [x(1), ..., x(t), u(1), ..., u(t)] <= uvar"
+function build_QP_JuMP_model(Q,R,A,B, nt;
+        x0   = [],
+        lvar = [], 
+        uvar = [],
+        Qf  = [])
+
+    if size(Qf)[1] ==0
+        Qf = copy(Q)
+    end
+
+    ns = size(Q)[1] # Number of states
+    nu = size(R)[1] # Number of inputs
+
+    NS = 1:ns # set of states
+    NT = 1:nt # set of times
+    NU = 1:nu # set of inputs
+
+
+
+    model = Model(Ipopt.Optimizer) # define model
+
+    @variable(model, x[NS, NT]) # define states 
+    @variable(model, u[NU, NT]) # define inputs
+
+    # Bound states/inputs
+    if length(lvar) > 0
+        for j in NT
+            for i in NS
+                @constraint(model, x[i,j] >= lvar[(j-1)*ns + i])
+            end
+
+            for i in NU
+                @constraint(model, u[i,j] >= lvar[ns*nt + (j-1)*nu + i])
+            end
+
+        end
+    end
+
+
+    if length(uvar) > 0
+        for j in NT
+            for i in NS
+                @constraint(model, x[i,j] <= uvar[(j-1)*ns + i])
+            end
+
+            for i in NU
+                @constraint(model, u[i,j] <= uvar[ns*nt + (j-1)*nu + i])
+            end
+
+        end
+    end
+
+    if length(x0) != 0
+        for i in NS
+            @constraint(model, x[i,1] == x0[i])
```

Comment:
to be consistent with what we have for QMs, let's replace this to bound constraints for variables

Reply:
In other words, I should not treat $x_0$ as a variable, correct? Right now, I have been treating it as a variable, but it makes sense that I should remove that.

Reply:
ah, sorry. what I meant is "fixing" variables https://jump.dev/JuMP.jl/stable/manual/variables/